apps/ledger-live-mobile/src/components/RequiresBLE/AndroidRequiresBluetoothPermissions.tsx

```diff
@@ -1,118 +1,45 @@
-import { Button, Flex, Link, Text } from "@ledgerhq/native-ui";
-import React, { ReactNode, useCallback, useEffect, useState } from "react";
-import { useTranslation } from "react-i18next";
-import { AppState, Linking, PermissionsAndroid } from "react-native";
-import { useIsMounted } from "../../helpers/useIsMounted";
-import BottomModal from "../BottomModal";
-import {
-  bluetoothPermissions,
-  checkBluetoothPermissions,
-  requestBluetoothPermissions,
-  RequestMultipleResult,
-} from "./androidBlePermissions";
-import BluetoothDisabled from "./BluetoothDisabled";
+import React, { ReactNode } from "react";
+import BluetoothPermissionDenied from "./BluetoothPermissionDenied";
+import { useAndroidBluetoothPermissions } from "./hooks/useAndroidBluetoothPermissions";
 
-const { RESULTS } = PermissionsAndroid;
+type Props = {
+  children?: ReactNode | undefined;
+  hasBackButtonOnDenied?: boolean;
+  openSettingsOnErrorButton?: boolean;
+};
 
 /**
- * Renders an error if location is required & not available,
- * otherwise renders children
+ * Renders an error if bluetooth is required & its associated permissions are not
+ * set. Otherwise renders children.
+ *
+ * Should only be used for Android.
+ *
+ * @param hasBackButtonOnDenied If true, the back button will be displayed on the permission denied screen. Defaults to false.
+ * @param openSettingsOnErrorButton Used for debug purposes. If true, on a permission denied, pressing the button on
```

Comment:
if it's for debug purposes only can we change the name of the prop? like prefix it with "debugOnlyDONOTUSE" or some thing like that which will jump to any reviewer's eyes so it's not used in real user situations haha

Reply:
I mean. it could be use in a scenario where a consumer of this component wants to force opening the user settings when pressing the button when there is an error

But i don't really see a case currently where we would want to force this

So i guess yeah we should rename it with `debugOnlyOpenSettingsOnErrorButton` ?

Reply:
🤷‍♂️ yeah or if you think it's not debug only, "forceOpenSettingsOnErrorButton", just something to make a consumer 2nd guess if he really wants to use this prop.
To be honest not super important 😆

Reply:
i like `forceOpenSettingsOnErrorButton` , i'll use this 👌